contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,231 @@
+</box>
+
+### Templates and Components
+
+The UI of an Angular app is defined in _templates_ which contain a mix of HTML code and HTML-like [custom tags](https://angular.io/guide/elements). The template for the above example (defined in `person.component.html`) is given below:
+
+``` html
+<h1>About Me!</h1>
+<h2>Hello World! My name is {person.name}</h2>
+<div>
+<label>NAME: <input [(ngModel)]="person.name"/></label>
+</div>
+```
+  
+The templates used in Angular are _declarative_ because they describe _what_ the view should look like rather than give instructions on _how_ to construct the view. In the above example, we declare what we want to display (i.e., name) and Angular renders the view. A noteworthy feature about the UI rendered by Angular is its _dynamic behavior_.
+You will notice that as you change the name in the text box, the name rendered on the app is also updated. This is because the UI and the application data are automatically synchronized by Angular as explained [here](https://angular.io/guide/template-syntax#binding-syntax-an-overview).
+
+Angular organizes the instructions (written in TypeScript) that describe _how_ the view must rendered, into _Components_. Given below is the component file `person.component.ts` of the `PersonComponent` :  
+
+```typescript
+@Component({
+  selector: 'app-person',
+  templateUrl: './person.component.html',
+  styleUrls: ['./person.component.css']
+})
+export class PersonComponent {
+  person: Person = {
+    name: 'John Doe'
+  };
+  constructor() { }
+}
+```
+
+The `@Component` tag is used to mark a component. It provides the metadata that determines how the component is processed during runtime.
+
+* The use of `selector: app-person` is how Angular identifies this component in a template. This is an example of a custom tag used in Angular templates. The following code constitutes the template of `app.component.html`.
+
+    ```html
+    <h1>Person Data</h1>
+    <app-person></app-person>
+    <hr/>
+    ```
+
+    Angular identifies the `<app-person>` tag, instantiates the `PersonComponent` and renders the view defined by the template `./person.component.html`.
+
+* The `person.name` used in the template `./person.component.html`, refers to the name property defined in the above component.
+
+### Dependency Injection
+
+Now let us look at what **Dependency Injection**(DI) means. Dependencies are services that a class needs in order to perform its function. We can visualize this with the help of an example.
+
+<center>
+<img src="dependency.png" width="60%">
+
+_Figure 1. What is a dependency?_ <sup>[source](https://medium.freecodecamp.org/a-quick-intro-to-dependency-injection-what-it-is-and-when-to-use-it-7578c84fa88f)</sup>
+</center>
+
+Dependency injection is a way of providing a class with the required services. The official website claims that the DI framework increases the "efficiency and modularity" of an app. Given below is a an injectable service class.
+ 
```

Comment:
You can actually get rid of this sentence as the point about "efficiency and modularity" is mentioned again later.
>The official website claims that the DI framework increases the "efficiency and modularity" of an app.